### PR TITLE
Prefactor; cleanup unused ES methods, add tests

### PR DESCRIPTION
**Pull request by sravfeyn**
## Product Description
This is backend cleanup and prefactor to add new test coverage in preparation for https://dimagi-dev.atlassian.net/browse/USH-3611

## Technical Summary

The deleted methods were added for ICDS use and were never removed after their usage was cleaned up. 

I have also pulled out a query that's getting used in Admin user list report so that tests can be added.

## Feature Flag

NA

### Safety story

Tested the admin report that's modified by this change.

### Automated test coverage

Added more, removed unnecessary.

### QA Plan

Not needed

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/es/users.py [hunk 1]
```diff
@@ -195,22 +194,10 @@ def user_ids(user_ids):
...
-            filters.term('domain_memberships.location_id', location_id)
-        ),
-    )
-
-
 def location(location_id):
     # by any assigned-location primary or not
     return filters.OR(
         filters.AND(mobile_users(), filters.term('assigned_location_ids', location_id)),
-        # todo; this actually doesn't get applied since the below field is not indexed
```
Comment (sravfeyn) [inline corehq/apps/es/users.py:213]: This is not true anymore, the filter does return webusers as well. Added a test for that https://github.com/dimagi/commcare-hq/pull/33694/files#diff-e9b28d016f1b2ed92a19743a7fd359714d363a993c485dd3fb34ddfac1e17430R125

File: corehq/apps/hqadmin/reports.py [hunk 2]
```diff
@@ -240,15 +240,8 @@ def rows(self):
             ]
 
     def _users_query(self):
-        query = (user_es.UserES()
-                 .remove_default_filters()
-                 .OR(user_es.web_users(), user_es.mobile_users()))
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py:245]: I think this might actually be necessary to filter out unknown users

**Review by esoergel — APPROVED** [review]
lots of nice cleanup here!

**Review by esoergel — APPROVED** [review]
At some point we should probably make an "exclude_unknown_users" filter, maybe even as a default

File: corehq/apps/hqadmin/reports.py (hunk 2 above)
Comment (sravfeyn) [reply corehq/apps/hqadmin/reports.py:245]: https://github.com/dimagi/commcare-hq/pull/33694/commits/1769f9f111bb281a222a4972b86581fb2abbf98f

**Pull request merged**